libfreerdp/core/client.c

```diff
@@ -303,7 +303,7 @@ UINT freerdp_channels_attach(freerdp* instance)
 	WINPR_ASSERT(instance->context->settings);
 
 	channels = instance->context->channels;
-	hostname = instance->context->settings->ServerHostname;
+	hostname = freerdp_settings_get_string(instance->context->settings, FreeRDP_ServerHostname);
 	hostnameLength = strlen(hostname);
```

Comment:
Hum perhaps we can add a WINPR_ASSERT here or `strlen`  will segfault or replace the call with `_strlen`

Reply:
@hardening true, added a `WINPR_ASSERT` and limit the hostname length to `MAX_PATH` (`URL` are less and file paths most likely too)